llvm/include/llvm/RemoteCachingService/RemoteCacheServer.h

```diff
@@ -1,49 +1,47 @@
-//===-- RemoteCacheServer.h -------------------------------------*- C++ -*-===//
+//===- llvm/RemoteCachingService/RemoteCacheServer.h ------------*- C++ -*-===//
 //
 // Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef LLVM_TOOLS_REMOTECACHETEST_REMOTECACHESERVER_H
-#define LLVM_TOOLS_REMOTECACHETEST_REMOTECACHESERVER_H
+#ifndef LLVM_REMOTECACHINGSERVICE_REMOTECACHESERVER_H
+#define LLVM_REMOTECACHINGSERVICE_REMOTECACHESERVER_H
 
 #include "llvm/ADT/StringRef.h"
 #include "llvm/Support/Error.h"
 
 namespace llvm {
-namespace remote_cache_test {
+namespace cas {
 
 class RemoteCacheProvider;
 class RemoteCacheServer;
-
-/// Returns a gRPC server for the remote caching protocol, with \p CacheProvider
-/// doing the actual work of storing and retrieving the data.
-RemoteCacheServer
-createServer(StringRef SocketPath,
-             std::unique_ptr<RemoteCacheProvider> CacheProvider);
```

Comment:
That doesn't work well because I want to hide all the grpc protocol from public headers (those in include/llvm), but in order to do anything to that `std::unique_ptr` inside the function signature, we need a concrete type for `RemoteCacheProvider` but that type has grpc dependency.

I decide that I will keep the header clean and we can figure out how to separate them cleanly when we have a second provider.